Run this on Android

KINGDOMWARE opened the pull request:
Hello, I would like to know how to use GPIOs in Android system. The final goal is to make some APK (using Android Studio etc) to control GPIO`s

awesometic committed:
ODROID-C4: New support for ODROID-C4

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Signed-off-by: Luke Go <sangch.go@gmail.com>
Change-Id: I0fd2de97d95bb8bf71cc1975081177571b1e8301

tobetter committed:
ODROID: fix to build Debian package without running 'build' script

Signed-off-by: Dongjin Kim <tobetter@gmail.com>

tobetter committed:
ODROID: fix absolute path problem after building

Signed-off-by: Dongjin Kim <tobetter@gmail.com>

joshua-yang committed:
wiringPi: Add copyright in gpio help messages

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

joshua-yang committed:
wiringPi: People: Add Hardkernel team

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

joshua-yang committed:
wiringPi: newVersion: Fix for adding minor-fixed version

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

joshua-yang committed:
wiringPi: Bump up version to 2.44-1.1

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

xiane committed:
android: Make valide to link wiringPi library from vendor.

joshua-yang committed:
wiringPi: Add wiringPiISRCancel function

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

joshua-yang committed:
wiringPi: Bump up version to 2.44-2.0

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

xiane committed:
android: pthread: replace pthread_cancel.

- android doesn't support pthread_cancel. so use pthread_kill instead
  pthread_cacnel.

xiane committed:
android: replace using 'gpio' to direct node touching.

hhk7734 committed:
ODROID-N2: Add new support for hardware PWM

Servo motors are typically controlled 1000 us ~ 2000 us pulse widths.
So default PWM period was set to 2500us.

xiane committed:
wiringPi: pthread: add mutex lock to avoid null pointer error.

- In some case, ISR callback function pointer can be null pointer untill pthread
  is not killed. So add mutext to avoid this problem and exit it.

Signed-off-by: Luke Go <luke.go@hardkernel.com>

xiane committed:
wiringPi: Remove global pinPass.

    - pinPass value has a possibilit to wrong pinvalue pass.

Signed-off-by: Luke Go <luke.go@hardkernel.com>

hhk7734 committed:
Add wiringPiSPISetupInterface

wiringPi did not support device selection.
When adding an SPI device or  SPI CS, a user did not use wiringPiSPI.h.

If using wiringPiSPISetupInterface, a user can select an SPI device.

joshua-yang committed:
ODROID-N2: Add static keyword to the new PWM functions

To avoid unforeseen collision, add static keyword to the functions.
And did some codes clean-up to fit into the exist code style.

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

joshua-yang committed:
wiringPi: Bump up version to 2.44-3.0

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

hhk7734 committed:
Add i2c Block read and write

wiringPiI2CReadBlock:
S Addr Wr [A] reg [A]
           S Addr Rd [A] [buff] A [buff] A ... A [buff] NA P

wiringPiI2CWriteBlock:
S Addr Wr [A] reg [A] buff [A] buff [A] ... [A] buff [A] P

hhk7734 committed:
Add pullup/down mode to pinMode

ex)
pinMode(pin, INPUT_PULLUP);
pinMode(pin, INPUT_PULLDOWN);

hhk7734 committed:
Add PWM comments for N2

hhk7734 committed:
Add #ifndef ANDROID to pinMode

The PWM default clock/range settings
  conflict with Android things settings.

joshua-yang committed:
wiringPi: Bump up version to 2.44-4.0

Signed-off-by: Joshua Yang <joshua.yang@hardkernel.com>

awesometic committed:
wiringPi: Fix typo

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Change author information

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Fix non-root user who has gpiomem cannot use gpio binary properly

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Bump up version to 2.44-4.1

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Add udev rules to correct /dev/gpiomem permission

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
build: Tune make jobs

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Add readall option '--all' and refactor readall.c

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Change readall output width to 79 for compatibility

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Bump up version to 2.44-5.0

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Fix for parsing /proc/cpuinfo properly

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Bump up version to 2.44-5.1

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Fix for parsing /proc/cpuinfo on ODROID-C

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Bump up version to 2.44-5.2

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Add a udev rules file for meson-gpiomem for ODROID-C/C2 boards

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
wiringPi: Bump up version to 2.44-5.3

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
WiringPi: Get model name from Device Tree if not able from cpuinfo

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: I0ca6497da3a9b0123b552fe7c245f08915d69a39

awesometic committed:
wiringPi/Makefile: Remove legacy files when uninstallation

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: Ic126b5d27b92f265151b29d287e2dcce3fcbc815

awesometic committed:
build: Clean up build script and do uninstall/clean before install

This because to make sure removing legacy files that could occurs
library conflicts.

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: Id795969241ba925104d374b99b496c799b9ed1df

awesometic committed:
debian: libwiringpi2: Remove previous built header files after installing

To make sure avoiding conflicts with legacy files.

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: I6c2ef994ce0b4c76603c89f76091d7cf8cfdca5e

awesometic committed:
WiringPi: Bump up version to 2.44-6.0

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: Ic8a3e66f58018b965ecbe893f6d318f0e1c9f9f1

awesometic committed:
WiringPi: Ignore case when identifying a board

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: Ia9edd472026f45ad9b9b923f652ca27d43d51f19

awesometic committed:
gpio: Add options for readall command in help text

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
Change-Id: I10a582184e46b15c6387b21ce3bd810a1c90d524

awesometic committed:
WiringPi: Make it does not configure /dev/gpiomem during deb package build

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

awesometic committed:
WiringPi: Bump up version to 2.44-6.1

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>

hhk7734 committed:
Fix issue where 'wiringPiISR' can't run without sudo

If using gpio command without sudo after changing group of gpio* from
'root' to 'gpio', the command failed. So, deleted the part using
the command.

https://github.com/hhk7734/oduino/issues/7

awesometic committed:
Merge pull request #29 from hhk7734/master

Fix issue where 'wiringPiISR' can't run without sudo

awesometic committed:
WiringPi: Bump up version to 2.44-6.2

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>

awesometic committed:
WiringPi: Improve logic of getting board information

Now WiringPi doesn't require editted /proc/cpuinfo if a proper model
name is defined in target board's device tree.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I94b6b9595577546d785f018258746032bf9a9494

awesometic committed:
WiringPi: Remove files that seem won't be used or already have not been used

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I1189f950d0e51cbc4c569b1e7a1182dc5f8fe762

awesometic committed:
WiringPi: Android: Apply removes to Android.bp

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I7255c36857185ed18811695ce5ad88dafe5d27ab

awesometic committed:
WiringPi: Remove unused codes from newVersion script

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Id85bd9d589c4b7e176f594cce6505e72e4765027

awesometic committed:
WiringPi: Minor changes

- Fix compile warning
- Remove pwmSetMode function that is not used
- Change the name of changing drive functions
- Aggressively use of WiringPi setupCheck() function
- Move usingGpiomem variable into libodroid structure
- etc...

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I355556975cedc686885f3b91d7f1a4ec02f1e232

awesometic committed:
Add definition _GNU_SOURCE to remove warnings to do with implicit functions

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I02489bde9498c3acb22af381bd4852e5a667c677

awesometic committed:
Bring devLib/lcd* back

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ifcc4338c95d185b730e87542a5c053a6906c86cb

awesometic committed:
WiringPi: Bump up version to 2.44-7.0

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: If50894454b6ea118e0ab6699816648c197261869

awesometic committed:
Revert "wiringPi: newVersion: Fix for adding minor-fixed version"

This reverts commit 07cc4620e0ccc1cce8732f7a418be73445784624.

Change-Id: I9023d16ddfe808fccb74b58981933091e6917c1f

awesometic committed:
newVersion: Remove unused codes

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I50da462320edefe4cf552718d0a457b91142f50b

awesometic committed:
WiringPi: Bump up version to 3.0

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ia3586cad02d45bc34bf7dc6409c1eae09fe50af6

awesometic committed:
debian: Add explicit package dependency

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I6c840db8b188148bdce303b7b4ab0110f0d54644

awesometic committed:
debian: Fix typo

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I6ab68c20633549bb3f13aab592862377912d4971

awesometic committed:
WiringPi: Bump up version to 3.1

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ieab47180823f028b7c67ed56e24e6b72ae6d7f8c

xiane committed:
WiringPi: Android: Add N2Plus to ModelNames list.

- Add N2Plus to default N2 string.

Change-Id: I0a04d57ec1ffcb6ca63b78275df52243abccb02e

awesometic committed:
WiringPi: Bring back all of the stuffs I removed to do with peripherals

Imported from the official WiringPi repository, version 2.60.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I82dc00ab9f038f9bc4c6a9c4ca3affe5a5d33ae1

awesometic committed:
Revert "WiringPi: Android: Apply removes to Android.bp"

This reverts commit e5a99034ba750a3043f78e5dfc84ee1658932ec4.

Change-Id: Iae25108e053224e69a470b508c1026feb6d63c4c

awesometic committed:
WiringPi: Bump up version to 3.2

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I31b75e0019b747950308da756b65d871031a5d28

xiane committed:
WiringPi: Android: Redefine some function that's not supported.

- Some api are not supported or redefined by android libraries. So to
support them on the android, this patch redefined the apis. And, some
api are not allowed. So avoid including the package by the problem.

Change-Id: Icaf2716a82dd98fc3c4fc3c597d4e67f3df8d755

awesometic committed:
WiringPi: Check Device Tree first to recognize and change the referenced DT path

As of this quotes from the /sys/firmware/devicetree documents.

Userspace must not use the /sys/firmware/devicetree/base path directly,
but instead should follow /proc/device-tree symlink.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ib6cd8694cf3b429297e5e7154387c1fb18716600

awesometic committed:
WiringPi: Bump up version to 3.3

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I39e300fe2808ac807df860a3f9ffc058d4968a07

awesometic committed:
WiringPi: Bump up version to 3.4

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>

codewalkerster committed:
WiringPi: Android: Add HC4 to ModelNames list.

 - Add N2Plus to default N2 string.

Change-Id: I151c2e61a3df2b1a9db62699559648b46ef52009

awesometic committed:
WiringPi: Implement storing current operating board's kernel version

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I663d1df07e88ebae4e975afc2ebae6122d5918e9

awesometic committed:
WiringPi: Add showing current kernel version to the help text

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I8c1bb3e900ea2ddfa497b8949f7e2909666b3f5b

awesometic committed:
WiringPi: Remove utsname things from each board's file

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I3b4be4eac57c31867a1a1b380eef26b0866da903

awesometic committed:
WiringPi: Add cmpKernelVersion and update the kernelVerion structure

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ia8431e26578f007fd98d152c4915ce3caba07fb9

awesometic committed:
WiringPi: Update in response to the latest regulations about I2C/SPI nodes

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ia6992332b157a6b078f5d1f67b280d2fc3a09774

awesometic committed:
Odroid-XU4: Use cmpKernelVersion to determine ADC node

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I64a138c85592f75b0689e1c93c18511f966202ad

awesometic committed:
WiringPi: Bump up version to 3.5

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I2b81a719d75fe9ce28ed25aea3c215da59c287b1

awesometic committed:
WiringPi: Remove patch string in kernel number and add whole release string

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ie1510c9b855d81cbbaee07a794bb584a70ebc53d

awesometic committed:
Odroid-C4: Correct ADC nodes and change their access number as to the Odroid convention

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I60db52cdf1ff4137b75b7336796a0fbd93eb7998

awesometic committed:
WiringPi: Bump up version to 3.6

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I0f2b9c1f08a8e8fc9575c17601b616b93eafd073

awesometic committed:
Odroid-C4: Add base number to GPIO_H pad pins

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ic99455f1d7e8b3ebb4a412dc6fd0b82da1e663a3

awesometic committed:
Odroid-C1: Fix to keep compatibility with Odroid-C series

which are not having proper model name like "Hardkernel ODROID-C1".

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ifcb1923435f2e7c1500d9ce14bd84b1e7ae0060d

awesometic committed:
Odroid-C1/C2: Add I2C support for mainline kernel

This patch makes it uses /dev/i2c-0 node if its kernel version is above 4.y.z.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I2d8d30f019465eb883b323d73e81a1ce086e85d8

awesometic committed:
WiringPi: Bump up version to 3.7

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ided1c49e7aa259a9d1d6b92d96d3b3e605a865ad

codewalkerster committed:
Merge "WiringPi: Android: Add HC4 to ModelNames list."

awesometic committed:
Revert "WiringPi: Android: Add HC4 to ModelNames list."

This reverts commit f787046a6306075764d8a3a5c17be2cfc9e6fc0e.

awesometic committed:
WiringPi: Add support for Odroid-HC4

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: If64c712eef681a2a9aa4b7828d95b612a1de91db

awesometic committed:
Bump up version to 3.8

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>

xiane committed:
WiringPi: Android: Add odroidhc4.c to compile on the android platform.

Change-Id: I36451f7c5c2ca99506e666f49a3d04a3f981310a

awesometic committed:
Bump up version to 3.9

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>

awesometic committed:
Fix build error for gcc 10

Thanks to @jens1 on the Odroid forum for notifying us

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I0c1658246e92b4c1b135be002cb365bd46c8e764

tobetter commented:
Which board, ODROID-C4/N2/XU4, are you going to run the Android with?

KINGDOMWARE commented:
> Which board, ODROID-C4/N2/XU4, are you going to run the Android with?

Now I have BananaPI R1, but I planning Rasp PI4 or ASUS TInker2

I need the unified-manual to convert libraries to any desired GPIO map (Rasp be-like)

Banana R1 has Android 4.2, ASUS has Android 10 etc

tobetter commented:
> > Which board, ODROID-C4/N2/XU4, are you going to run the Android with?
> 
> Now I have BananaPI R1, but I planning Rasp PI4 or ASUS TInker2
> 
> I need the unified-manual to convert libraries to any desired GPIO map (Rasp be-like)
> 
> Banana R1 has Android 4.2, ASUS has Android 10 etc

Sorry, this repository is old and not updated for long time. Also, I have no intention to do for other boards.

KINGDOMWARE commented:
> Sorry, this repository is old and not updated for long time. Also, I have no intention to do for other boards.

I say thanks if you show me the method for your boards, then I will try to adopt this to another

tobetter commented:
> > Sorry, this repository is old and not updated for long time. Also, I have no intention to do for other boards.
> 
> I say thanks if you show me the method for your boards, then I will try to adopt this to another

You can refer to this branch.
https://github.com/hardkernel/wiringpi/tree/ODROID-XU3/4

tobetter committed:
Rework to build by autoconf/automake and new Debian scripts

Building in local machine:
    $ sh autogen.sh
    $ ./configure
    $ make
    $ sudo make install

Building Debian package:
    $ debuild -uc -us -b

Signed-off-by: Dongjin Kim <tobetter@gmail.com>
Change-Id: I76faa07c1586658d053a3e260ea231a19392f792

how2flow committed:
ODROID-M1: New support for ODROID-M1

Change-Id: I8691f53540d566d00945fe3c7b6fb2f5f317cc1f
Signed-off-by: Steve Jeong <jkhpro1003@gmail.com>

xiane committed:
Check build MODE to check gpiod fw.

Some Android doesn't support gpiod frameworks. So check MODE to include
gpiod fw. It should be determined before building.

Signed-off-by: Luke go <sangch.go@gmail.com>
Change-Id: I5b4627c86b4b083622822576c5afa0dd72894faf

how2flow committed:
Add include headers & change linux default lib: gpiod

Signed-off-by: jkhgit <jkhpro1003@gmail.com>
Change-Id: Ie43c001db0b4eefa55925a246ca4af4ae6abbf46

xiane committed:
ODROID: Add __attribute__ for avoding unused parameters.

Signed-off-by: Luke go <sangch.go@gmail.com>
Change-Id: I4ff66f9f12f336635b7ee58215fb0f4053d343e8

how2flow committed:
ODROID-N2/C4: fix compile error at http://ppa.launchpad.net/hardkernel/ppa/ubuntu

wiringPi/Makefile.am:50: error: blank line following trailing backslash
autoreconf: automake failed with exit status: 1
dh_autoreconf: error: autoreconf -f -i returned exit code 1
make: *** [debian/rules:6: build-arch] Error 25

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I29457b243d846bcaf08c1d9da5423066e777ca85

how2flow committed:
debian: update package version 3.13

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ic250c144a27c93b4da8942bb885692f0455544dc

how2flow committed:
debian: update package version 3.13.1

ODROID-COMMON: Add link libgpiod
ODROID-N2/C4: fix odroid-wiringpi postinst

server build error: link
./configure: line 3480: syntax error near unexpected token `LIBCRYPT,'
./configure: line 3480: `PKG_CHECK_MODULES(LIBCRYPT, libcrypt)'

local build error: link
/usr/bin/ld: undefined reference to `gpiod_line_set_value' ...

local build error: odroid-wiringpi.postinst error
cp: cannot stat '/tmp/odroid-wiringpi/rules.d/99-odroid-wiringpi-aml.rules': No such file or directory

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I0faf7710461ab487afdf02a8e5511ea2a6385f7e

how2flow committed:
debian: update package version 3.13.2

gpio readall error:
gpio pin value dumped when using command "gpio write $PIN $VALUE" and call "gpio readall"

ODROID-M1: Add udev rules for gpiomem

default: gpiomem

change machanism gpiomem or libgpiod
use or not -DDEVMEM in wiringPi/Makefile.am (Linux) Android.bp (Android)

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ie389c338e5082a94638d00983801f18fef1cbfaa

how2flow committed:
ODROID-N2/C4: add path of ADC in kernel 5.x

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I5abeaa0c863bb447e6a44efc9a1e348cb190ccf9

how2flow committed:
debian: update package version 3.13.3

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I59643fc4c7bad0c68229f8cda3fc431da85cdc48

how2flow committed:
ODROID-COMMON: [solve] syntax error near unexpected token `LIBCRYPT,'

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I1bce1ff782ba93bf76253a8df3dcca70457b5090

how2flow committed:
deb-versdion: Add epoch.

update package version 3.13.4

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I0c0443efcd8ab01273e107a8a34a15c04e8ae7be

how2flow committed:
debian: update package version 3.13.5

modify depends of libwiringpi2. it has itself as a depedency.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ie42ae9714584f24721521328dbc22d130ae525e7

how2flow committed:
debian: update package version 3.13.6

Issue: "DEVMEM" not defined in pip package.
Solve: define "DEVMEM" in m1 header file.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I5d6af496d78547971b03b77f02c6523dc5d20d3a

how2flow committed:
debian: update package version 3.14.0

ODROID-N2L: new support ODROID-N2L.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Id15ea8832dba1872a5bc90ebc426d182a7b8b2e6

how2flow committed:
debian: update package version 3.13.7

ODROID-M1: support PWM OUTPUT. (PWM1, PWM2)

TODO:
	support PWM9.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I657e61f33c8bdc948e58a04e3729db69166c9e45

how2flow committed:
debian: update package version 3.13.8

ODROID-M1: troubleshooting 2x pwm freq.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Idd13e496ef864986bd8c9c79a6ea6ae680b7e9ba

how2flow committed:
debian: update package version 3.13.9

ODROID-M1: pwm: modify setRange flag.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ib86f11be66eaaa33ec35a74f07188268f395348f

how2flow committed:
debian: update package version 3.13.10

ODROID-COMMON: remove dependency "libgpiod".

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I6531c8cca311a39522e9881719021c7eb61773da

how2flow committed:
debian: update package version 3.14.1

ODROID-COMMON: support hardware PWM with sysnode.
Support pwm with gpiomem mode. not sys mode.

ANDROID: Add Android.mk for android.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ic6005c0cb81586c9f945b9b6c75f904fa79e3915

how2flow committed:
ODROID-COMMON: Add function for input data to sysnode.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I7a1c6b82f8461b78aa5f9616531f9e2ae08c0fd3

how2flow committed:
debian/changelog: fix e-mail addresss

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I66969501ae81767ca2200ba8d86690338a368326

xiane committed:
ODROID-COMMON: inputToSysNode: Changed error log.

- Changed fopen error log for making easy to debug.

Signed-off-by: Luke Go <sangch.go@gmail.com>
Change-Id: Icf410b3aa0b6c4e732413a811054c585cade727f
Signed-off-by: steve.jeong <jkhpro1003@gmail.com>

how2flow committed:
ODROID-COMMON: pwm: improve pwm API & Update missing M1/N2 codes.

the missing commit: 8fceae1

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I1092e044a0e500944fc36e38c6d991e7e9bdde5c

how2flow committed:
debian: update package version 3.14.2

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I528cfc071c4e9ee26e4488d0a68ada62496f889f

how2flow committed:
libwiringpi-dev/depends: Add pkg-config.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I2809439436adf2223617a8e50cced0ada227cea5

how2flow committed:
ODROID-COMMON: remove exit to prevent system down during setup.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: I0b932277079c24c2c178f8c1ef8c79904dc2c08e

how2flow committed:
debian: update package version 3.14.3

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Iab2a516de0a8314fd39ea7bfb7b54784d8ba8c20

how2flow committed:
ODROID-COMMON: modify pwmSetRange to to call just once.

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Ie05850032dea81d03eacfdf21fea96006886585d

how2flow committed:
debian: update package version 3.14.4

Signed-off-by: steve.jeong <jkhpro1003@gmail.com>
Change-Id: Iecff8f4b4c2d937ef6abee858595e4b9d0e738e1

Commit:
change maintainer's e-mail and name.

change mail address gmail of "jkhpro1003" to "how2soft".

"jkh" means that is the initials of my korean name spelling.
I changed it because only Koreans don't use it.

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Ibf7ceef9b1085f706bb0ce82fcddfa2d13a16b9d

Commit:
debian: update package version 3.14.5

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: I34e352bd9007536a6c8724764cf28da7d0a84f7b

codewalkerster committed:
ODROID-XU3/4: Fix wiringPiISR.

If a pin is exported, ISR does not work on XU4.
We will clear pin before using with wiringPiISR function.

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Ibc747fe30fda20253eb77d9bbe5f6bcc8c2180b8

Commit:
debian: update package version 3.14.6

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Iec3ad0a30a9623feb96da774944bd4b7bd27fb10

Commit:
etc: rename script more generic.

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Ia3cdc818a9156a4aaad346e63efdea6e32ab864b

Commit:
debian: update package version 3.15.0

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Ia8921bc2b1f444412e0e8003ba499490b3883107

Commit:
wiringPi/example: Add i2c lcd example

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: I491a6aad8da6c606638f885921a7174017507ef1

Commit:
debian: update package version 3.15.1

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: Ib3acd042d211c90281a9933975234a6b6570f0f7

Commit:
wiringPi/examples: improve i2c-lcd.c

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: I8602dca171b3d681d5481b541e087bed47d2a761

Commit:
debian: update package version 3.15.2

Signed-off-by: Steve Jeong <how2soft@gmail.com>
Change-Id: I86064e5b6f41f2b119ba5c3d3c94f9979378387e

Commit:
The maintainer's email address has been changed.

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I111e25d014248274696c3b216851deaeab2b7f6e

Commit:
debian: update package version 3.15.3

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I078b9f7f0c8acbd446850396632f85e4698cc666

how2flow committed:
ODROID-M1S: New support for ODROID-M1S

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ib134094f7a35298b1274a60fbfc56a07a14ed965

how2flow committed:
ODROID-M1: fix pin of softTone and correct typos

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I2e5e4a0c992aa53df5b1ee54e3a13095f9fa8364

how2flow committed:
debian: update package version 3.15.4

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I67a7626e6a258f9116608e6763e786389565245f

how2flow committed:
debian: update package version 3.16.0

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ic7e72e6894df43ad62e00b9200c23df6b984c54d

how2flow committed:
ODROID-M1S: Supports Extra pins

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I48b505ea29ac10d1d9acbaed0ef99fc091ecd270

how2flow committed:
debian: update package version 3.16.1

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I7db24f9325f3c0a85b2abf379a8183ef63144f5f

how2flow committed:
ODROID-M2: New support for ODROID-M2

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I379018ae50e6c2c43f0e53d9bc5986a2090d7818

how2flow committed:
ODROID-M1/S: Rename vars more intuitively

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I4a0f0b571f3bf703940e51859eedd8d848ad169e

how2flow committed:
ODROID-M1S: Fix init function name

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I2f8bc62cf0b7bbde707b10e36e39e3a5a5c3045c

how2flow committed:
ODROID-M1/S: Change CALC

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ie0eefc480949dfb65ff58c1c2ec2ab0990a0d421

how2flow committed:
ODROID-M1/S: Remove PWM register addresses

The hardware pwm is used as the sys node.

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I01181a32aba992a11f41176b485f5472872f84a2

how2flow committed:
debian: update package version 3.16.2

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I4ed61b46ade6b8a2f702a8669b33d83840d86270

how2flow committed:
ODROID-M1/S: Refactoring to eliminate repeated operations, mechanisms for easy maintenance

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ieef76b76baa387454a174332168bd20b6d3e659b

how2flow committed:
debian: update package version 3.16.3

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I6756baefda6a49945fbf03790e4a0810164f9064

how2flow committed:
ODROID-M1/M1S: Remove gpiod

gpiod is supported through circuitpython platform.
odroid-wiringPi will operate faster gpio control as an advantage, so discard the gpiod.

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ibce3c53a1e0c1b6ad1c066b0966ec0d0dcb07e58

how2flow committed:
ODROID-M1/S: Rename/Update Read/Write Byte and DS LEVEL

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I4e803c6410efb4924473b17e6857848ee7edbe0a

how2flow committed:
ODROID-M1/S: Fix PWM mux error

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I17b12bf8762a5beb1ab149074ab27eaac4234967

how2flow committed:
debian: update package version 3.16.4

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: Ic3bf1cc6e2f0b8d46a9ff22c966780c634076e16

how2flow committed:
debian: update package version 3.17.0

Signed-off-by: Steve Jeong <steve@how2flow.net>
Change-Id: I7a4daa82410cc030884bf9468762b2cc502955bb

xiane committed:
ODROID-M2: Fix for build error.

Signed-off-by: xiane <sangch.go@gmail.com>
Change-Id: I62c2c7ae7cc1458fbccbac260374f0f1251b209a

shinys000114 committed:
ODROID-C5: Add odroid-c5 wiring-pi support

- Rev. 20250106
- Warning: Use only GPIO functions.

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I02623a284c047c6b5e485d28c9714ed8e65025df

shinys000114 committed:
ODROID: Change install path sbin to bin

Change-Id: Id9839e55cdc557e608af6f29d831d8c1bd8de9fa

shinys000114 committed:
ODROID-C5: Add pwm support

- Rev. 20250106

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I68581313c6bfe23b870380547e59249075aba3d0

shinys000114 committed:
ODROID-C5: Use wiringpi in sudo

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I42bf4afda5a5cb6fc3996c12cb1cb192e58157d6

shinys000114 committed:
ODROID-C5: increase analog read buffer size

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: Ic8b90557cf9585759559d8869590df0233ba8280

shinys000114 committed:
ODROID-C5: updates according to board revision REV 1.0 20250313

Change-Id: I1f91f06f5c6ccad767c6219fcbb190206fd571fe

shinys000114 committed:
ODROID-C5: pwm: fix "config pwm dtb overlays"

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I64840ede68cd71966d8696ccccf091e88e16f9b2

shinys000114 committed:
ODROID-C5: readall: fix pin name

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I6f7c39567e56e5558ab7cd16c2efe21bb29dc21f
(cherry picked from commit 45e673d119e53605ca973ced49332d1a47131f06)

shinys000114 committed:
ODROID-C5: readall: fix pin name

Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
Change-Id: I73553adfbe39f81b65d8d92ad83b42ab0007f408

shinys000114 committed:
ODROID: Update udev rules to change group from "odroid" to "gpio"

Change-Id: I7617617042f7bfb52c46ca7f5ba81c08ffcf4adc

shinys000114 committed:
debian: fix sudo-rs wildcard compatibility

Use a standalone trailing wildcard in the sudoers command arguments
to avoid warnings from sudo-rs while preserving PWM access.

/etc/sudoers.d/odroid-wpi-sudoers:2:31: wildcards are not allowed in command arguments
%odroid ALL = (ALL) NOPASSWD: /usr/bin/sh /etc/odroid-allowuser.sh*
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Change-Id: Ie198c5b14929b3a2f01a0f77663ae078cfdc652b
Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>

shinys000114 committed:
debian: change maintainer

Change-Id: I1222ea27cc0fc07be6751af3ac360ee3650803ed
Signed-off-by: YoungSoo Shin <shinys000114@gmail.com>
